src/Components/Server/test/Circuits/ComponentHubTest.cs

```diff
@@ -0,0 +1,209 @@
+                circuitIdFactory: circuitIdFactory,
+                circuitRegistry: circuitRegistry,
+                circuitHandleRegistry: circuitHandleRegistry,
+                logger: NullLogger<ComponentHub>.Instance);
+
+            // Here we mock out elements of the Hub that are typically configured
+            // by SignalR as clients connect to the hub.
+            var mockCaller = new Mock<IHubCallerClients>();
+            var mockClientProxy = new Mock<IClientProxy>();
+            mockCaller.Setup(x => x.Caller).Returns(mockClientProxy.Object);
+            hub.Clients = mockCaller.Object;
+            var mockContext = new Mock<HubCallerContext>();
+            mockContext.Setup(x => x.ConnectionId).Returns("123");
+            hub.Context = mockContext.Object;
+
+            return (mockClientProxy, hub);
+        }
+
+        private class TestCircuitHandleRegistry : ICircuitHandleRegistry
+        {
+            private bool circuitSet = false;
+
+            public CircuitHandle GetCircuitHandle(object circuitKey)
+            {
+                return null;
+            }
+
+            public CircuitHost GetCircuit(object circuitKey)
+            {
+                if (circuitSet)
+                {
+                    var serviceScope = new Mock<IServiceScope>();
+                    var circuitHost = TestCircuitHost.Create(
+                        serviceScope: new AsyncServiceScope(serviceScope.Object));
+                    return circuitHost;
+                }
+                return null;
+            }
+
+            public void SetCircuit(object circuitKey, CircuitHost circuitHost)
+            {
+                circuitSet = true;
+                return;
+            }
+        }
+
+        private class TestServerComponentDeserializer : IServerComponentDeserializer
+        {
+            public bool TryDeserializeComponentDescriptorCollection(string serializedComponentRecords, out List<ComponentDescriptor> descriptors)
+            {
+                descriptors = default;
+                return true;
+            }
+        }
+
+        private class TestCircuitFactory : ICircuitFactory
+        {
+            public TestCircuitFactory(
+            IServiceScopeFactory scopeFactory,
+            ILoggerFactory loggerFactory,
+            CircuitIdFactory circuitIdFactory,
+            IOptions<CircuitOptions> options) { }
+
+            // We override the default `CreateCircuitHostAsync` so we can mock around
+            // the work of initializing a host and all its service dependencies.
+            public async ValueTask<CircuitHost> CreateCircuitHostAsync(
+                IReadOnlyList<ComponentDescriptor> components,
+                CircuitClientProxy client,
+                string baseUri,
+                string uri,
+                ClaimsPrincipal user,
+                IComponentApplicationStateStore store)
+            {
+                await Task.Delay(0);
```

Comment:
If this is still required, could you add a comment to clarify why? It raises alarms otherwise.

For example would it work to make this method not `async` and return `Task.FromResult(circuitHost)` instead?

Reply:
It's not needed. This was just me typing the fewest characters possible to make things go away in dev.